lib/solid/parser.ex

```diff
@@ -96,18 +96,27 @@ defmodule Solid.Parser.Base do
 
       argument = choice([value, field])
 
-      opening_object = string("{{")
-      closing_object = string("}}")
-
       opening_tag = string("{%")
       closing_tag = string("%}")
 
+      opening_objects =
+        choice([
+          string("{{-") |> replace(true) |> tag(:trim_previous),
+          string("{{") |> replace(false) |> tag(:trim_previous)
+        ])
+
+      closing_objects =
+        choice([
+          string("-}}") |> replace(true) |> tag(:trim_next),
+          string("}}") |> replace(false) |> tag(:trim_next)
```

Comment:
I think if you change the `tag` calls for `unwrap_and_tag` you don't end up with a list and you can simply use

`trim_next = Keyword.get(data, :trim_next)`